src/ORM/Query.php

```diff
@@ -1126,7 +1125,8 @@ protected function _execute(): iterable
             return $this->_results;
         }
 
-        $results = $this->execute()->fetchAll(StatementInterface::FETCH_TYPE_ASSOC);
+        /** @var array $results */
```

Comment:
This is a hack since we know Database\Query actually uses an array.

Reply:
We can look into changing the arguments which accept this results to `iterable` instead of `array`.

Reply:
Updated to convert to `array` before passing to `loadExternal()`. Will work on results sets separately.